src/dynamic-import/import-defer-script-code-valid.case

```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2025 Igalia, S.L. All rights reserved.
+// Copyright (C) 2018 Leo Balter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+desc: import.defer() can be used in script code
+template: syntax/valid
+features: [import-defer]
+---*/
+
+//- setup
+// This is still valid in script code, and should not be valid for module code
+// https://tc39.github.io/ecma262/#sec-scripts-static-semantics-lexicallydeclarednames
+var smoosh; function smoosh() {}
+
```

Comment:
I did not realize that we had automatic `setup` and `teardown` substitution variables in the test generator! (They are documented; I just never noticed.)

Anyway, not having `flags: [module]` in the test seems like it's sufficient for ensuring it's executed as script code, so is this really needed?

Reply:
I copied this from the `import.source` tests (these `.case` files are basically a copy-paste of those). It is also use in the `import` tests. I'm going to remove it from the `import.defer` tests, but probably it should then also be removed from those, since it's indeed useless (unless we want to catch broken test262 runners).

Reply:
Sure, would you like to make a follow-up PR for that?

Reply:
I'll open an issue assigned to myself for now :P